mk/spksrc.common.mk

```diff
@@ -119,3 +119,6 @@ version_le = $(shell if printf '%s\n' "$(1)" "$(2)" | sort -VC ; then echo 1; fi
 version_ge = $(shell if printf '%s\n' "$(1)" "$(2)" | sort -VCr ; then echo 1; fi)
 version_lt = $(shell if [ "$(1)" != "$(2)" ] && printf "%s\n" "$(1)" "$(2)" | sort -VC ; then echo 1; fi)
 version_gt = $(shell if [ "$(1)" != "$(2)" ] && printf "%s\n" "$(1)" "$(2)" | sort -VCr ; then echo 1; fi)
+
+# GCC version
+version_gcc = $(shell $$(sed -n -e '/TC_ENV += TC_GCC/ s/.*= *//p' $(PWD)/$(subst arch,work,$(MAKECMDGOALS))/tc_vars.mk))
```

Comment:
@hgy59 I'd like to bring your attention to this function helper so we can finally make usage of the GCC version from any spk.  See use-case in python310 Makefile which allows to capture `comcerto2k-7.0` exception cleanly.

Reply:
My findings so far:
- you don't have `TC_ENV` nor `TC_GCC` available before the target `$(TC_VARS_MK)` is built.
- you can modify makefile variables in a custom `PRE_DEPEND_TARGET` that depends on `$(TC_VARS_MK)` with $(eval )

``` 
.PHONY: python310_pre_depend
python310_pre_depend: $(TC_VARS_MK)
	@$(MSG) "Evaluate dependencies for GCC version $(TC_GCC)"
	$(eval WHEELS_BUILD_ARGS += $(shell if "$(TC_GCC)" != "5.0" && printf '%s\n' "$(TC_GCC)" "5.0" | sort -VC ; then echo --b2-args=cxxstd=11 ; fi))
	@$(MSG) "WHEELS_BUILD_ARGS=$(WHEELS_BUILD_ARGS)"
``` 

- ensure the `$(TC_VARS_MK)` is built before
- version comparision migth be improved (but `version_lt` is not ment for shell environment)

Reply:
Your findings where pretty much the route I was originally leaning towards but the relatively simplistic method I've moved to uses the `TC_GCC` version variable from the `tc_vars.mk` and works quite well.

It actually solves a long-standing issue I had with:
```
ifeq ($(call version_ge, ${TCVERSION}, 7.0),1)
# fix compilation with newer compilers
ADDITIONAL_CXXFLAGS += -D_GLIBCXX_USE_C99_MATH -D_GLIBCXX_HAVE_OBSOLETE_ISNAN -D_GLIBCXX_HAVE_OBSOLETE_ISINF
endif
```
That can then be converted into (and takes into consideration `comcerto2k-7.0`):
```
ifeq ($(call version_ge, $(call version_gcc), 7.5),1)
# fix compilation with newer compilers
ADDITIONAL_CXXFLAGS += -D_GLIBCXX_USE_C99_MATH -D_GLIBCXX_HAVE_OBSOLETE_ISNAN -D_GLIBCXX_HAVE_OBSOLETE_ISINF
endif
```